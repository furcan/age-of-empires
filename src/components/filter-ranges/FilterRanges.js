import { FiInfo } from 'react-icons/fi';

import './FilterRanges.scss';

function FilterRanges(props) {
  const {
    onChangeHandler,
    costs,
    valueWood,
    valueFood,
    valueGold,
    setMouseStateWood,
    rangeWoodHandler,
    setMouseStateFood,
    rangeFoodHandler,
    setMouseStateGold,
    rangeGoldHandler,
  } = props;

  return (
    <div className="ranges-wrapper">
      <div className="range-group">
        <div className="range-check">
          <input type="checkbox" id="Wood" name="useWood" onChange={(event) => onChangeHandler(event)} />
          <label htmlFor="Wood">Wood</label>
        </div>

        <div className="range-slider">
          <input type="range" id="WoodRange" min="0" max="200" step="1" value={valueWood} onChange={(event) => rangeWoodHandler(event)} onMouseDown={() => setMouseStateWood('down')} onMouseUp={() => setMouseStateWood('up')} disabled={!costs.useWood} />
          <label htmlFor="WoodRange">
            <FiInfo />
            <span>{`0 - ${valueWood}`}</span>
          </label>
        </div>
      </div>


      <div className="range-group">
        <div className="range-check">
          <input type="checkbox" id="Food" name="useFood" onChange={(event) => onChangeHandler(event)} />
          <label htmlFor="Food">Food</label>
        </div>

        <div className="range-slider">
          <input type="range" id="FoodRange" min="0" max="200" step="1" value={valueFood} onChange={(event) => rangeFoodHandler(event)} onMouseDown={() => setMouseStateFood('down')} onMouseUp={() => setMouseStateFood('up')} disabled={!costs.useFood} />
          <label htmlFor="FoodRange">
            <FiInfo />
            <span>{`0 - ${valueFood}`}</span>
          </label>
        </div>
      </div>

      <div className="range-group">
        <div className="range-check">
          <input type="checkbox" id="Gold" name="useGold" onChange={(event) => onChangeHandler(event)} />
          <label htmlFor="Gold">Gold</label>
        </div>

        <div className="range-slider">
          <input type="range" id="GoldRange" min="0" max="200" step="1" value={valueGold} onChange={(event) => rangeGoldHandler(event)} onMouseDown={() => setMouseStateGold('down')} onMouseUp={() => setMouseStateGold('up')} disabled={!costs.useGold} />
          <label htmlFor="GoldRange">
            <FiInfo />
            <span>{`0 - ${valueGold}`}</span>
          </label>
        </div>
      </div>

    </div>
  );
}

export default FilterRanges;
