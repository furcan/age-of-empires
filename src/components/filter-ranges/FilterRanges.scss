@import "../../styles/_variables";

.ranges-wrapper {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;

  .range-group {
    width: 100%;
    display: flex;
    flex-direction: column;
    margin: 0 0 25px;

    &:last-child {
      margin: 0;
    }

    @media (min-width: $breakpoint-mobile) {
      width: 33.3333%;
      padding: 10px;
      margin: 0;
    }

    .range-check {
      width: 100%;
      display: flex;
      flex-wrap: wrap;
      align-items: center;
      margin: 0 0 20px;

      input[type="checkbox"] {
        cursor: pointer;
        width: 24px;
        height: 24px;
      }

      label {
        cursor: pointer;
        margin-left: 10px;
        font-size: 16px;
        line-height: 1;
        font-weight: 600;
      }
    }

    .range-slider {
      width: 100%;
      display: flex;
      flex-wrap: wrap;
      flex-direction: column;
      align-items: center;

      label {
        transition: opacity 0.25s ease-in-out;
        cursor: default;
        margin: 5px 0 0;
        font-size: 12px;
        line-height: 1;
        font-weight: 500;
        color: $gray;
        display: flex;
        flex-wrap: wrap;
        align-items: center;

        i,
        svg {
          opacity: 0.25;
          margin-right: 4px;
        }
      }

      input[type="range"] {
        transition: opacity 0.25s ease-in-out;
        cursor: pointer;
        width: 100%;
        margin: 0;

        &[disabled] {
          opacity: 0.4;

          + label {
            opacity: 0.4;
          }
        }
      }
    }
  }
}
