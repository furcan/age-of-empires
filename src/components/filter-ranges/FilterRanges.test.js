
import renderer from 'react-test-renderer';

import FilterRanges from './FilterRanges';

it('The FilterRanges component has been rendered as expected.', () => {
  const tree = renderer.create(<FilterRanges />).toJSON();
  expect(tree).toMatchSnapshot();
});
