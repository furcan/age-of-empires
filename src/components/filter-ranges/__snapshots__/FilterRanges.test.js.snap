// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FilterRanges component has been rendered as expected. 1`] = `
<div
  className="ranges-wrapper"
>
  <div
    className="range-group"
  >
    <div
      className="range-check"
    >
      <input
        id="Wood"
        name="useWood"
        type="checkbox"
      />
      <label
        htmlFor="Wood"
      >
        Wood
      </label>
    </div>
    <div
      className="range-slider"
    >
      <input
        disabled={true}
        id="WoodRange"
        max="200"
        min="0"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step="1"
        type="range"
      />
      <label
        htmlFor="WoodRange"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="8"
            y2="8"
          />
        </svg>
        <span>
          0 - undefined
        </span>
      </label>
    </div>
  </div>
  <div
    className="range-group"
  >
    <div
      className="range-check"
    >
      <input
        id="Food"
        name="useFood"
        type="checkbox"
      />
      <label
        htmlFor="Food"
      >
        Food
      </label>
    </div>
    <div
      className="range-slider"
    >
      <input
        disabled={true}
        id="FoodRange"
        max="200"
        min="0"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step="1"
        type="range"
      />
      <label
        htmlFor="FoodRange"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="8"
            y2="8"
          />
        </svg>
        <span>
          0 - undefined
        </span>
      </label>
    </div>
  </div>
  <div
    className="range-group"
  >
    <div
      className="range-check"
    >
      <input
        id="Gold"
        name="useGold"
        type="checkbox"
      />
      <label
        htmlFor="Gold"
      >
        Gold
      </label>
    </div>
    <div
      className="range-slider"
    >
      <input
        disabled={true}
        id="GoldRange"
        max="200"
        min="0"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        step="1"
        type="range"
      />
      <label
        htmlFor="GoldRange"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="8"
            y2="8"
          />
        </svg>
        <span>
          0 - undefined
        </span>
      </label>
    </div>
  </div>
</div>
`;
